### PR TITLE
Minor rendering improvements

### PR DESCRIPTION
**Pull request by zzzi2p**
This is a collection of minor changes we use in I2P. Some of it is a matter of taste or may be different than what jrobin did. Other changes omit output when the alpha or stroke is 0, which is useful for SVG output but is more efficient regardless. Please review and cherrypick what you like and skip what you don't.

- Change u to micron in legend to match value axis
- Combine drawRules() and drawSpans() to only iterate through plotElements once
- Only clip and reset in drawRulesAndSpans() if we output something
- Move title up a little for better padding below
- Position long titles to clip only on right side
- Don't draw arrowheads if disabled by color alpha of 0
- Don't draw ticks if disabled by stroke width of 0
- Don't draw three legend color boxes on top of each other unless all have some transparency
- Replace NaN with 0 in legend values
- Output milli values as 0.xxx rather than xxx m
- Add day number to time axis day names
- Javadoc warning fixes

### CODE REVIEW

File: src/main/java/org/rrd4j/graph/RrdGraphGenerator.java [hunk 1]
```diff
@@ -624,11 +638,17 @@ private void drawLegend() {
...
-                        worker.fillRect(x + 1, y - box + 1, box - 2, box - 2, ((LegendText) c).legendColor);
+                        Paint bc = gdef.getColor(ElementsNames.back);
+                        Paint lc = ((LegendText) c).legendColor;
+                        boolean bt = bc.getTransparency() != Transparency.OPAQUE;
+                        boolean lt = lc.getTransparency() != Transparency.OPAQUE;
+                        // no use drawing unless both the two on top have some transparency
+                        if (bt && lt)
+                            worker.fillRect(x + 1, y - box + 1, box - 2, box - 2, gdef.getColor(ElementsNames.canvas));
+                        // no use drawing unless the one on top has some transparency
+                        if (lt)
```
Comment (fbacchella) [inline src/main/java/org/rrd4j/graph/RrdGraphGenerator.java:649]: Can wrap all `if` in a block `{}` even one liner ?

File: src/main/java/org/rrd4j/graph/ValueAxis.java [hunk 2]
```diff
@@ -151,7 +154,10 @@ boolean draw() {
                         }
                     }
                     else {
-                        if (fractional) {
+                        if (im.symbol == 'm') {
```
Comment (fbacchella) [inline src/main/java/org/rrd4j/graph/ValueAxis.java:157]: I don’t get this one.

File: src/main/java/org/rrd4j/graph/ValueScaler.java [hunk 3]
```diff
@@ -16,6 +16,9 @@ class ValueScaler {
     }
 
     Scaled scale(double value, boolean mustRescale) {
+        // avoid NaN in legend
```
Comment (fbacchella) [inline src/main/java/org/rrd4j/graph/ValueScaler.java:19]: NaN is not 0, so it should be kept in the legend.

File: src/main/java/org/rrd4j/graph/RrdGraphGenerator.java (hunk 1 above)
Comment (zzzi2p) [reply src/main/java/org/rrd4j/graph/RrdGraphGenerator.java:649]: of course. you can do that if you're cherry-picking to match your code style.

File: src/main/java/org/rrd4j/graph/ValueScaler.java (hunk 3 above)
Comment (zzzi2p) [reply src/main/java/org/rrd4j/graph/ValueScaler.java:19]: You're right of course, but for our less-technical users in multiple languages, 0 makes more sense than NaN. You can skip it if you don't want this change.

**Comment by fbacchella** [comment]
I will download the patch file and apply it manually instead of doing a merge request, so I can tweak it to suits my needs. I wonder if I can add some your changes as arguments to rrdef. Some of them are really matter of taste and there is absolutely no "best solution".

**Pull request closed without merging**